### ISSUE TITLE
combining ampersand as class of same element selector doesn't work

### ISSUE THREAD

**Issue by Zxurian** [sass/sass#1760]
when trying to reference a parent selector using an element to include parent selector as a class, it errors out. 
## Expected css compilation:

``` css
.stamp {
    color: rgb(255, 0, 0);
    background-color: rgb(0, 0, 255);
}
a.stamp {
    color: rgb(0, 255, 0);
    background-color: rgb(0, 0, 255);
}
```
## What I've tried

``` sass
.stamp {
    color: rgb(255, 0, 0);
    background-color: rgb(0, 0, 255);
    a& {
        color: rgb(0, 255, 0);
    }
}
```

sassmeister reports as `invalid selector after a on line 4 at column 3`

``` sass
.stamp {
    color: rgb(255, 0, 0);
    background-color: rgb(0, 0, 255);
    a.& {
        color: rgb(0, 255, 0);
    }
}
```

sassmeister reports as `property "a" must be followed by a ':' on line 4 at column 3`
## Working alternatives

The only way I can accomplish what I'm looking for right now that I can see is

``` sass
.stamp {
    color: rgb(255, 0, 0);
    background-color: rgb(0, 0, 255);
}
a.stamp {
    @extend .stamp;
    color: rgb(255, 0, 0);
}
```

which seems unecessary to me given that the `&` can be used as parent references in other contexts. (`&:hover`, `a &`, etc...)

**Comment by cimmanon** [comment]
Duplicate of https://github.com/sass/sass/issues/1369

**Comment by Zxurian** [comment]
thank you, apparenly I was searching the wrong terms.

The last comment says it was addressed in 3.4. Yet using sassmeister with compiler set to 3.4.14 (was on 3.2 by default) I can't seem to get it to function using just the ampersand, although it does function using the `@at-root` method.

Is there any documentation that describes the change in syntax when using ampersand as a compound selector instead of by itself?
